tests/integration-tests/src/test/java/org/apache/activemq/artemis/tests/integration/replication/SharedNothingReplicationTest.java

```diff
@@ -0,0 +1,388 @@
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicInteger;
+
+public class SharedNothingReplicationTest {
+    private static final Logger logger = Logger.getLogger(SharedNothingReplicationTest.class);
+
+    @Rule
+    public TemporaryFolder brokersFolder = new TemporaryFolder();
+
+    private SlowMessagePersister slowMessagePersister;
+
+    @Before
+    public void setUp() throws Exception {
+        slowMessagePersister = new SlowMessagePersister();
+        CoreMessagePersister.theInstance = slowMessagePersister;
+    }
+
+    @After
+    public void tearDown() throws Exception {
+        if (slowMessagePersister != null) {
+            CoreMessagePersister.theInstance = slowMessagePersister.persister;
+        }
+    }
+
+    @Test
+    public void testReplicateFromSlowLive() throws Exception {
+        // start live
+        Configuration liveConfiguration = createLiveConfiguration();
+        ActiveMQServer liveServer = ActiveMQServers.newActiveMQServer(liveConfiguration);
+        liveServer.start();
+
+        Wait.waitFor(() -> liveServer.isStarted());
+
+        CoreMessagePersister.theInstance = SlowMessagePersister._getInstance();
+
+        final CountDownLatch replicated = new CountDownLatch(1);
+
+        ServerLocator locator = ServerLocatorImpl.newLocator("tcp://localhost:61616");
+        locator.setCallTimeout(60_000L);
+        locator.setConnectionTTL(60_000L);
+        locator.addClusterTopologyListener(new ClusterTopologyListener() {
+            @Override
+            public void nodeUP(TopologyMember member, boolean last) {
+                logger.infof("nodeUP fired last=%s, live=%s, backup=%s", last, member.getLive(), member.getBackup());
+                if (member.getBackup() != null) {
+                    replicated.countDown();
+                }
+            }
+
+            @Override
+            public void nodeDown(long eventUID, String nodeID) {
+
+            }
+        });
+
+        final ClientSessionFactory csf = locator.createSessionFactory();
+        ClientSession sess = csf.createSession();
+        sess.createQueue("slow", RoutingType.ANYCAST, "slow", true);
+        sess.close();
+        Executor sendMessageExecutor = Executors.newCachedThreadPool();
+
+
+        // let's write some messages
+        int i = 0;
+        final int j = 50;
+        final CountDownLatch allMessageSent = new CountDownLatch(j);
+        while (i < 5) {
+            sendMessageExecutor.execute(() -> {
+                try {
+                    ClientSession session = csf.createSession(true, true);
+                    ClientProducer producer = session.createProducer("slow");
+                    ClientMessage message = session.createMessage(true);
+                    // this will make journal's append executor busy
+                    message.putLongProperty("delay", Long.getLong("message.property.delay",500L));
```

Comment:
Where? how this property is affecting the semantic of the server. I didn't find it anywhere.

Reply:
Never mind.. I see.. it's on your extended persister...


I don't think we need a system property.. the test can just keep it constant at 500L